examples/iotivity_examples/README.md

```diff
@@ -0,0 +1,314 @@
+           inet6 addr: ff02::1:ffbd:1f80/128  scope: local [multicast]
+           inet6 addr: ff02::2/128  scope: local [multicast]
+           inet6 addr: ff02::1a/128  scope: local [multicast]
+           inet6 addr: ff03::158/128  scope: global [multicast]
+
+client_oic: continue discovery
+Outgoing message to [ff03:0000:0000:0000:0000:0000:0000:0158]:5683
+client_oic: continue discovery
+...
+ipadapter: got server request
+Incoming message from [fe80:0000:0000:0000:241c:c8ff:fe14:3d79]:56789
+ipadapter: waiting for server requests...
+client_oic: Discovery done
+client_oic: Ready...
+LED_RED_ON
+```
+Once the resource is discovered, the client registers as an Observer of the resource and it switches on its LED as notification of Discovery Completed.
+From this point it will send a PUT request every second.
+Client Output:
+```
+client_oic: Sent PUT request
+Outgoing message to [fe80:0000:0000:0000:241c:c8ff:fe14:3d79]:56789
+ipadapter: got server request
+Incoming message from [fe80:0000:0000:0000:241c:c8ff:fe14:3d79]:56789
+ipadapter: waiting for server requests...
+client_oic: PUT_light:
+client_oic: PUT response OK
+ipadapter: got server request
+Incoming message from [fe80:0000:0000:0000:241c:c8ff:fe14:3d79]:56789
+ipadapter: waiting for server requests...
+client_oic: OBSERVE_light: key state, value 0
+```
+Server output:
+```
+Incoming message from [fe80:0000:0000:0000:a4ab:89ff:febd:1f80]:56789
+ipadapter: waiting for server requests...
+server_oic: PUT request
+server_oic: key: state value: 0
+server_oic: LED0 is OFF
+LED_RED_OFF
+Outgoing message to [fe80:0000:0000:0000:a4ab:89ff:febd:1f80]:56789
+server_oic: GET request
+server_oic: Light state 0
+Outgoing message to [fe80:0000:0000:0000:a4ab:89ff:febd:1f80]:56789
+```
+TAPs interfaces can be easily deleted. Go to `/dist/tools/tapsetup` and type
+```
+$ sudo ./tapsetup -d
+```
+
+###<a name="sc_pput_samr21"></a> Server and Client (Periodic PUT) - SAMR21-XPRO target
+Now, we reproduce the previous scenario using two [SAMR21-XPRO][4] nodes.
+Connect your nodes, go to `/examples/iotivity-examples/server` and check the list of USB-connected nodes by typing:
+```
+$ make list-ttys
+```
+The output will be similar to
+```
+/sys/bus/usb/devices/2-1.3: Atmel Corp. EDBG CMSIS-DAP serial: 'ATML2127031800001234', tty(s): ttyACM0
+/sys/bus/usb/devices/2-1.4: Atmel Corp. EDBG CMSIS-DAP serial: 'ATML2127031800004321', tty(s): ttyACM1
+```
+We will use Serial Numbers in order to identify the designed node during the compilation phase.
+Now, we compile the server
+```
+$ make flash BOARD=samr21-xpro SERIAL=server_node_serial
+```
+then we open the serial connection
+```
+$ make term BOARD=samr21-xpro SERIAL=server_node_serial
+```
+The server starts and it is waiting for incoming requests.
+Now,  open a new terminal window, go to `/examples/iotivity-examples/client` and type
+```
+$ make flash BOARD=samr21-xpro SERIAL=client_node_serial
```

Comment:
I noticed that if the client was compiled beforehand this only works with first 
`make clean all`
and then 
`make flash BOARD=samr21-xpro SERIAL=...`
else you end up with client code instead of server code.
(I have not checked yet in details why, or if this is the only scenario where things get mixed up).
Does it ring a bell?